### ISSUE TITLE
Azure DevOps can't access aks private api address

### ISSUE THREAD

**Issue by Joris29** [preddy727/aksprivatecluster#1]
Hi i'm setting up a private aks cluster and i want to use azure pipelines to build and deploy to my aks cluster but i can't seem to connect to the private api address of the aks cluster from the azure devops pipelines.

![aks](https://user-images.githubusercontent.com/25951077/81274592-c8de0b80-9050-11ea-807b-6c39582b8474.png)

**Comment by preddy727** [comment]
Are you using a self hosted agent that is in the same vnet as the private endpoint that was created to the kubectl control plane?

**Comment by Joris29** [comment]
@preddy727 hi i managed to figure it i'll close this.

**Comment by surya2155** [comment]
@Joris29 How did u resolve this?? I tried to link my self hosted agent Vnet to the private azure DNS (using the vnet linking ) which already has 'A' rule for the private K8 api endpoint address.
Could you let me know what have you done to resolve the issue!!!

**Comment by surya2155** [comment]
@preddy727 any suggestions here?

**Comment by Joris29** [comment]
@surya2155 i decided to use a public cluster but there is a solution:
https://github.com/MicrosoftDocs/azure-docs/issues/54304#issuecomment-628037012

**Comment by preddy727** [comment]
> @preddy727 any suggestions here?

As mentioned above, use a self hosted agent and create a private endpoint in the devops subnet to the aks resource id.

**Comment by ttsakpc** [comment]
and for authenticating to AKS (via your self hosted agent) see this blog post:

https://colinsalmcorner.com/azure-pipelines-for-private-aks-clusters/

TL;DR you use kubeconfig connection method and choose "Save without Verification" (since Azure Devops cannot access the AKS cluster to verify, but the self hosted agent will be able to access it using this kubeconfig)
